api/src/routes/api/spec/stp.spec.ts

```diff
@@ -207,57 +208,61 @@ describe('stp', () => {
           await createFile(sasProgramPath, sampleSasProgram)
           await createFile(jsProgramPath, sampleJsProgram)
 
-          await request(app)
-            .get(`/SASjsApi/stp/execute?_program=${programPath}`)
-            .auth(accessToken, { type: 'bearer' })
-            .send()
-            .expect(200)
-
-          expect(ProcessProgramModule.processProgram).toHaveBeenCalledWith(
-            expect.anything(),
-            expect.anything(),
-            expect.anything(),
-            expect.anything(),
-            expect.anything(),
-            expect.anything(),
-            RunTimeType.JS,
-            expect.anything(),
-            undefined
-          )
+          await makeRequestAndAssert(programPath, 200, RunTimeType.JS)
         })
 
         it('should execute sas program when js program is not present but sas program exists', async () => {
           const programPath = path.join(testFilesFolder, 'program')
           const sasProgramPath = path.join(filesFolder, `${programPath}.sas`)
           await createFile(sasProgramPath, sampleSasProgram)
 
-          await request(app)
-            .get(`/SASjsApi/stp/execute?_program=${programPath}`)
-            .auth(accessToken, { type: 'bearer' })
-            .send()
-            .expect(200)
-
-          expect(ProcessProgramModule.processProgram).toHaveBeenCalledWith(
-            expect.anything(),
-            expect.anything(),
-            expect.anything(),
-            expect.anything(),
-            expect.anything(),
-            expect.anything(),
-            RunTimeType.SAS,
-            expect.anything(),
-            undefined
-          )
+          await makeRequestAndAssert(programPath, 200, RunTimeType.SAS)
         })
 
         it('should throw error when both sas and js programs do not exist', async () => {
           const programPath = path.join(testFilesFolder, 'program')
 
-          await request(app)
-            .get(`/SASjsApi/stp/execute?_program=${programPath}`)
-            .auth(accessToken, { type: 'bearer' })
-            .send()
-            .expect(400)
+          await makeRequestAndAssert(programPath, 400)
+        })
+      })
+
+      describe('with runtime py and sas', () => {
+        beforeAll(() => {
+          process.runTimes = [RunTimeType.PY, RunTimeType.SAS]
+        })
+
+        beforeEach(() => {
+          jest.resetModules() // it clears the cache
+          setupMocks()
+        })
+
+        afterEach(async () => {
+          jest.resetAllMocks()
+          await deleteFolder(path.join(filesFolder, testFilesFolder))
+        })
+
+        it('should execute python program when both python and sas program are present', async () => {
+          const programPath = path.join(testFilesFolder, 'program')
```

Comment:
I can see lots of duplications for each runtime, so it makes sense to create utils

Reply:
updated